tests/framework/installer/ceph_installer.go

```diff
@@ -444,57 +424,53 @@ func (h *CephInstaller) GetNodeHostnames() ([]string, error) {
 	return names, nil
 }
 
-// InstallRook installs rook on k8s
-func (h *CephInstaller) InstallRook(namespace, storeType string, usePVC bool, storageClassName string,
-	mon cephv1.MonSpec, startWithAllNodes bool, rbdMirrorWorkers int, skipOSDCreation bool, rookVersion string) (bool, error) {
+func (h *CephInstaller) InstallRook() (bool, error) {
+
+	if h.settings.RookVersion != VersionMaster {
+		// make sure we have the images from a previous release locally so the test doesn't hit a timeout
+		assert.NoError(h.T(), h.k8shelper.GetDockerImage("rook/ceph:"+h.settings.RookVersion))
+	}
+
+	assert.NoError(h.T(), h.k8shelper.GetDockerImage(h.settings.CephVersion.Image))
 
 	ctx := context.TODO()
 	var err error
 	k8sversion := h.k8shelper.GetK8sServerVersion()
 
-	logger.Infof("Installing rook on k8s %s", k8sversion)
+	logger.Infof("Installing rook on K8s %s", k8sversion)
 
-	startDiscovery := true
-	onamespace := namespace
+	startDiscovery := false
 	// Create rook operator
-	if h.useHelm {
+	if h.settings.UseHelm {
 		// disable the discovery daemonset with the helm chart
-		settings := "enableDiscoveryDaemon=false,image.tag=master"
-		startDiscovery = false
-		err = h.CreateRookOperatorViaHelm(namespace, settings)
+		startDiscovery = true
+		err = h.CreateRookOperatorViaHelm("enableDiscoveryDaemon=true,image.tag=master")
 		if err != nil {
-			logger.Errorf("Rook Operator not installed ,error -> %v", err)
+			logger.Errorf("helm install failed. %v", err)
```

Comment:
So we log the error and return it? I wonder if the logger is needed here, it doesn't hurt though

Reply:
hmm this should really be an assert, otherwise it will cascade to other issues, will fix.